build: (initial?) failure to build xproto on Alpine

```bash
# podman run -it alpine
apk update && apk add pkgconf patch python3 cmake g++ bash curl gcc git make && \
	git clone https://github.com/bitcoin/bitcoin && \
	cd bitcoin && \
	make -C depends qt -j13
```

This will fail with
```bash
Staging xproto...
make[1]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
Making install in specs
make[2]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
Making install in SIAddresses
make[3]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[4]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[4]: Nothing to be done for 'install-exec-am'.
make[4]: Nothing to be done for 'install-data-am'.
make[4]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[3]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[3]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[4]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[4]: Nothing to be done for 'install-exec-am'.
make[4]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[3]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[2]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[2]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[3]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[3]: Nothing to be done for 'install-exec-am'.
 ./install-sh -c -d '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/lib/pkgconfig'
 ./install-sh -c -d '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11'
 ./install-sh -c -d '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11'
mkdir: can't create directory '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11': File exists
make[3]: *** [Makefile:508: install-xprotoHEADERS] Error 1
make[3]: *** Waiting for unfinished jobs....
 /usr/bin/install -c -m 644 xproto.pc '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/lib/pkgconfig'
 /usr/bin/install -c -m 644 Xfuncproto.h Xpoll.h '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11'
make[3]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[2]: *** [Makefile:838: install-am] Error 2
make[2]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[1]: *** [Makefile:535: install-recursive] Error 1
make[1]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make: *** [funcs.mk:342: /bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/.stamp_staged] Error 2
make: Leaving directory '/bitcoin/depends'
```

If you restart the build it then picks up and continues:
```bash
make -C depends/ qt
make: Entering directory '/bitcoin/depends'
Staging xproto...
make[1]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
Making install in specs
make[2]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
Making install in SIAddresses
make[3]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[4]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[4]: Nothing to be done for 'install-exec-am'.
make[4]: Nothing to be done for 'install-data-am'.
make[4]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[3]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs/SIAddresses'
make[3]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[4]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[4]: Nothing to be done for 'install-exec-am'.
make[4]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[3]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[2]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/specs'
make[2]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[3]: Entering directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[3]: Nothing to be done for 'install-exec-am'.
 ./install-sh -c -d '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11'
 /usr/bin/install -c -m 644 Xfuncproto.h Xpoll.h '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11'
 ./install-sh -c -d '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/lib/pkgconfig'
 /usr/bin/install -c -m 644 xproto.pc '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/lib/pkgconfig'
 ./install-sh -c -d '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11'
 /usr/bin/install -c -m 644 ap_keysym.h DECkeysym.h HPkeysym.h keysymdef.h keysym.h Sunkeysym.h Xalloca.h Xarch.h Xatom.h Xdefs.h XF86keysym.h Xfuncs.h X.h Xmd.h Xosdefs.h Xos.h Xos_r.h Xproto.h Xprotostr.h Xthreads.h Xw32defs.h XWDFile.h Xwindows.h Xwinsock.h '/bitcoin/depends/work/staging/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec/bitcoin/depends/aarch64-unknown-linux-musl/include/X11'
make[3]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[2]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
make[1]: Leaving directory '/bitcoin/depends/work/build/aarch64-unknown-linux-musl/xproto/7.0.31-7e4871430ec'
Postprocessing xproto...
Caching xproto...
```

--- a/depends/packages/xproto.mk
+++ b/depends/packages/xproto.mk
@@ -21,6 +21,8 @@ define $(package)_build_cmds
   $(MAKE)
 endef
 
+# mkdir detection is broken on Alpine. Set MKDIRPROG to ensure we always
+# use "mkdir -p", and avoid parallelism issues during install.
 define $(package)_stage_cmds
-  $(MAKE) DESTDIR=$($(package)_staging_dir) install
+  $(MAKE) MKDIRPROG="mkdir -p" DESTDIR=$($(package)_staging_dir) install
 endef

